cfg/gnu.cfg

```diff
@@ -25,15 +25,45 @@
     <alloc init="false" buffer-size="malloc">pvalloc</alloc>
     <dealloc>free</dealloc>
   </memory>
-  <define name="bswap_16(x)" value="((unsigned short int) ((((x) &gt;&gt; 8) &amp; 0xff) | (((x) &amp; 0xff) &lt;&lt; 8)))"/>
+  <function name="bswap_16">
+    <noreturn>false</noreturn>
+    <use-retval/>
+    <returnValue type="uint16_t"/>
```

Comment:
Could you test such configuration:

    <returnValue type="uint16_t">((arg1 &amp; 0xf0) &gt;&gt; 8) | ((arg1 &amp; 0xf) &lt;&lt; 8)</returnValue>

Then we should for instance detect such bugs:

    x = 100 / bswap_16(0);  // <- should warn about division by zero

Reply:
@danmar Ah ok, will do. I wasn't aware that this is possible. It doesn't seem to be documented anywhere and I didn't notice it when taking the other `.cfg` files as reference.

Reply:
@danmar To make the expression behave correctly, I have to cast the argument to `uint16_t`, but it seems that this breaks the value analysis. With

```<returnValue type="uint16_t">(((uint16_t)arg1 &amp; 0xff00u) &gt;&gt; 8) | (((uint16_t)arg1 &amp; 0x00ffu) &lt;&lt; 8)</returnValue>```

the division by zero in your test case `x = 100 / bswap_16(0);` is not recognized. But if I use instead

```<returnValue type="uint16_t">((arg1 &amp; 0xff00u) &gt;&gt; 8) | ((arg1 &amp; 0x00ffu) &lt;&lt; 8)</returnValue>```

it is.

Is this a known issue? I think I would commit the first variant and add your test case to `test/cfg/gnu.c`, but comment it out as TODO.

Reply:
I have the feeling that casts are not supported in these expressions yet .. well maybe we should have a ticket for that. A TODO comment for that case sounds reasonable.